core/storage/collection/gae_models_test.py

```diff
@@ -0,0 +1,208 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for collection models."""
+from constants import constants
+from core.domain import collection_domain
+from core.domain import collection_services
+from core.platform import models
+from core.tests import test_utils
+
+(collection_models,) = models.Registry.import_models(
+    [models.NAMES.collection])
+
+
+class CollectionModelUnitTest(test_utils.GenericTestBase):
+    """Test the CollectionModel class."""
+
+    def test_get_collection_count(self):
+        collection = collection_domain.Collection.create_default_collection(
+            'id', title='A title',
+            category='A Category', objective='An Objective')
+        collection_services.save_new_collection('id', collection)
+
+        num_collections = (
+            collection_models.CollectionModel.get_collection_count())
+        self.assertEquals(num_collections, 1)
+
+
+class CollectionCommitLogEntryModelUnitTest(test_utils.GenericTestBase):
+    """Test the CollectionCommitLogEntryModel class."""
+
+    def test_get_all_non_private_commits(self):
+        private_commit = collection_models.CollectionCommitLogEntryModel.create(
+            'id', None, 'commiter_id', 'username', 'msg', 'create', {},
+            constants.ACTIVITY_STATUS_PRIVATE, False)
+        public_commit = collection_models.CollectionCommitLogEntryModel.create(
+            'id', None, 'commiter_id', 'username', 'msg', 'create', {},
+            constants.ACTIVITY_STATUS_PUBLIC, False)
+        private_commit.collection_id = 'a'
+        public_commit.collection_id = 'b'
+        private_commit.put()
+        public_commit.put()
+        commits = (
+            collection_models.CollectionCommitLogEntryModel
+            .get_all_non_private_commits(2, None, None))
+        self.assertEquals(1, len(commits[0]))
```

Comment:
I don't think I quite follow this, sorry. I expected len(commits) to be 1 and maybe an equality check to compare commits[0] against an expected value? Let me know if I'm misunderstanding.

Reply:
Done.